include/AST/Types/Typekind.def

```diff
@@ -0,0 +1,42 @@
+/// Represents a structure type.
+/// Used to describe types defined as structures (e.g., structs in
+/// C/C++).
+TYPE_KIND(StructTyKind)
```

Comment:
You'd put StructTy here, the name of the type
You use NAME ## Kind in your enum macro to add the kind suffix